test/benchmarks/driver_bench/src/driver.mts

```diff
@@ -0,0 +1,194 @@
+);
+
+/**
+ * If you need to test BSON changes, you should clone, checkout and build BSON.
+ * run: `npm link` with no arguments to register the link.
+ * Then in the driver you are testing run `npm link bson` to use your local build.
+ *
+ * This will symlink the BSON into the driver's node_modules directory. So here
+ * we can find the revision of the BSON we are testing against if .git exists.
+ */
+export const MONGODB_BSON_REVISION = await (async () => {
+  const bsonGitExists = await fs.access(path.join(MONGODB_BSON_PATH, '.git')).then(
+    () => true,
+    () => false
+  );
+  if (!bsonGitExists) {
+    return 'installed from npm';
+  }
+  try {
+    return child_process
+      .execSync('git rev-parse --short HEAD', {
+        cwd: path.join(MONGODB_BSON_PATH),
+        encoding: 'utf8'
+      })
+      .trim();
+  } catch {
+    return 'unknown revision';
+  }
+})();
+
+export const MONGODB_CLIENT_OPTIONS = (() => {
+  const optionsString = process.env.MONGODB_CLIENT_OPTIONS;
+  let options = undefined;
+  if (optionsString?.length) {
+    options = JSON.parse(optionsString);
+  }
+  return { ...options };
+})();
+
+export const MONGODB_URI = (() => {
+  if (process.env.MONGODB_URI?.length) return process.env.MONGODB_URI;
+  return 'mongodb://127.0.0.1:27017';
+})();
+
+export function snakeToCamel(name: string) {
+  return name
+    .split('_')
+    .map((s, i) => (i !== 0 ? s[0].toUpperCase() + s.slice(1) : s))
+    .join('');
+}
+
+import type mongodb from '../../../../mongodb.js';
+export type { mongodb };
+
+const { MongoClient, GridFSBucket } = require(path.join(MONGODB_DRIVER_PATH));
+
+const DB_NAME = 'perftest';
+const COLLECTION_NAME = 'corpus';
+
+const SPEC_DIRECTORY = path.resolve(__dirname, '..', '..', 'driverBench', 'spec');
+
+export function metrics(test_name: string, result: number, count: number) {
+  return {
+    info: {
+      test_name,
+      // Args can only be a map of string -> int32. So if its a number leave it be,
+      // if it is anything else test for truthiness and set to 1 or 0.
+      args: Object.fromEntries(
+        Object.entries(MONGODB_CLIENT_OPTIONS).map(([key, value]) => [
+          key,
+          typeof value === 'number' ? value : value ? 1 : 0
+        ])
+      )
+    },
+    metrics: [
+      { name: 'megabytes_per_second', value: result },
+      // Reporting the count so we can verify programmatically or in UI how many iterations we reached
+      { name: 'count', value: count }
+    ]
+  } as const;
+}
+
+/**
+ * This class exists to abstract some of the driver API so we can gloss over version differences.
+ * For use in setup/teardown mostly.
+ */
+export class DriverTester {
+  private utilClient: mongodb.MongoClient;
+  public client: mongodb.MongoClient;
+  constructor() {
+    this.utilClient = new MongoClient(MONGODB_URI, MONGODB_CLIENT_OPTIONS);
+    this.client = new MongoClient(MONGODB_URI, MONGODB_CLIENT_OPTIONS);
+  }
+
+  private get ns() {
```

Comment:
Can we remove this and instead just use `collection`?  They're identical, and `ns` is confusing anyways because it's not actually a namespace, this returns a collection object.

Reply:
Yea I don't like it either 😅

I also realize the utilClient seems nice but I should really open and close it because it has monitoring overhead if we leave it running during the benchmark.